docs/docs/docs/01-core/data-transfer/01-engine.md

```diff
@@ -202,12 +202,24 @@ The engine includes a diagnostics reporter which can be used to listen for diagn
 Here is an example for creating a diagnostics listener:
 
 ```
-// your listener
-engine.diagnostics.onDiagnostic(async (data: GenericDiagnostic) => {
-  // handle the diagnostics, for example with custom logging
-});
+// listener function
+const diagnosticListener: DiagnosticListener = (data: GenericDiagnostic) => {
+  // handle the diagnostics event, for example with custom logging
+}
+
+// add a generic listener
+engine.diagnostics.onDiagnostic(diagnosticsListener);
+// OR
+engine.diagnostics.addListener('diagnostic', diagnosticListener);
+
+// add an error listener
+engine.diagnostics.addListener('diagnostic.error', diagnosticListener);
+
+// add a warning listener
+engine.diagnostics.addListener('diagnostic.warning', diagnosticListener);
```

Comment:
Where did you see this addListener? :thinking: 

From what I see you have access to:
```ts
function onDiagnostic(listener: DiagnosticListener): this;
function on<T extends DiagnosticKind>(kind: T, listener: DiagnosticListener<T>): this;

function report(diagnostic: Diagnostic): this;
```

Reply:
Oops, I was looking at the internal definition for the reoprter. Good catch.

Reply:
@Convly Fixed and added emitting events too.